src/core/main.c

```diff
@@ -1694,6 +1694,10 @@ static void initialize_coredump(bool skip_setup) {
         if (getpid_cached() != 1)
                 return;
 
+        /* Leave coredump settings as-is if systemd-coredump is unavailable */
+        if (access(ROOTLIBEXECDIR "/systemd-coredump", F_OK) < 0)
+                return;
```

Comment:
so if you check for the existance here, you might as well write the path you just checked into the sysctl, to turn it on.

also, this really needs some log output, explaining the situation.

Reply:
you mean replacing the sysctl setting? Sure can do that if that's preferred.
I guess the super correct solution would be to write the pattern only after `systemd-coredump.socket` is started, otherwise codedumps wouldn't work anyway, right?